deployment/service.tf

```diff
@@ -0,0 +1,107 @@
+#
+# ECS Resources
+#
+
+# ECS cluster is only a name that ECS machines may join
+resource "aws_ecs_cluster" "gtsite" {
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  name = "${var.stack_name}"
+}
+
+# Template for container definition, allows us to inject environment
+data "template_file" "ecs_gtsite_task" {
+  template = "${file("${path.module}/containers.json")}"
+
+  vars {
+    service_image            = "${var.service_image}"
+    static_image             = "${var.static_image}"
+  }
+}
+
+# Allows resource sharing among multiple containers
+resource "aws_ecs_task_definition" "gtsite" {
+  family                = "${var.stack_name}"
+  container_definitions = "${data.template_file.ecs_gtsite_task.rendered}"
+}
+
+# Defines running an ECS task as a service
+resource "aws_ecs_service" "gtsite" {
+  name                               = "${var.stack_name}"
+  cluster                            = "${aws_ecs_cluster.gtsite.id}"
+  task_definition                    = "${aws_ecs_task_definition.gtsite.family}:${aws_ecs_task_definition.gtsite.revision}"
+  desired_count                      = "${var.desired_instance_count}"
+  iam_role                           = "${var.ecs_service_role}"
+  deployment_minimum_healthy_percent = "0" # allow old services to be torn down
```

Comment:
I think it would be better not to hardcode these two settings, or at least increase them to more than zero, unless we're OK with a small amount of downtime on each deploy.